stan/math/prim/fun/cholesky_factor_constrain.hpp

```diff
@@ -85,6 +85,33 @@ cholesky_factor_constrain(const T& x, int M, int N, value_type_t<T>& lp) {
   return cholesky_factor_constrain(x_ref, M, N);
 }
 
+/**
+ * Return the Cholesky factor of the specified size read from the
+ * specified vector and increment the specified log probability
+ * reference with the log Jacobian adjustment of the transform.  A total
+ * of (N choose 2) + N + N * (M - N) free parameters are required to read
+ * an M by N Cholesky factor.
+ *
+ * @tparam Jacobian If true, incremented `lp` with the log Jacobian
+ * @tparam T A type inheriting from `Eigen::DenseBase` or a `var_value` with
+ *  inner type inheriting from `Eigen::DenseBase` with compile time dynamic rows
+ *  and 1 column.
+ * @param x Vector of unconstrained values
+ * @param M number of rows
+ * @param N number of columns
+ * @param lp Log probability that is incremented with the log Jacobian
+ * @return Cholesky factor
+ */
+template <bool Jacobian, typename T>
+inline auto cholesky_factor_constrain(const T& x, int M, int N,
+                                      scalar_type_t<T>& lp) {
+  if (Jacobian) {
```

Comment:
imo I like the `if (Jacobian)` for these just because it future proofs us a bit so that if we ever move to C++17 we can pretty much just do a copy replace for `if (Jacobian)` for `if constexpr (Jacobian)`